src/PythonQtPythonInclude.h

```diff
@@ -33,12 +33,33 @@
 #ifndef __PythonQtPythonInclude_h
 #define __PythonQtPythonInclude_h
 
+// Undefine macros that Python.h defines to avoid redefinition warning.
+#ifdef _POSIX_C_SOURCE
+#  undef _POSIX_C_SOURCE
+#endif
+
+#ifdef _POSIX_THREADS
+#  undef _POSIX_THREADS
+#endif
+
+#ifdef _XOPEN_SOURCE
+#  undef _XOPEN_SOURCE
+#endif
+
```

Comment:
@iakov - we havea problem with this code, as it undefines _POSIX_THREADS, which leads to errors due to missing defines in `pythread.h`. As far as I can see, `_POSIX_THREADS` is not defined in `Python.h`, it just is checked (in `pythread.h`) to see if POSIX threads are defined. It think it is defined in `pthread.h` or a related include. The same seems to be true for the other symbols.
I'm inclined to revert this specific change, though I may be misunderstanding something. Can you please elaborate on this?

Reply:
I'll make a PR to revert this to discuss this.